### PR TITLE
[CH-179] support prewhere on mergetree

### PR DESCRIPTION
**Pull request by liuneng1994**
## What changes were proposed in this pull request?

support prewhere on mergetree scan

## How was this patch tested?

unit tests

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)
test by CH[[185]]

### CODE REVIEW

**Comment by liuneng1994** [comment]
test with ch 185

**Comment by liuneng1994** [comment]
test with ch 185

**Comment by liuneng1994** [comment]
test with ch 185

**Comment by liuneng1994** [comment]
test with ch 185

**Comment by liuneng1994** [comment]
test with ch 185

**Comment by liuneng1994** [comment]
@deploy

**Comment by liuneng1994** [comment]
@deploy

**Comment by liuneng1994** [comment]
test

**Comment by liuneng1994** [comment]
test

**Comment by liuneng1994** [comment]
test with ch 185

**Comment by liuneng1994** [comment]
test with ch 185

**Comment by liuneng1994** [comment]
test with ch 185

File: backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHFilterExecTransformer.scala [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    TransformContext(inputAttributes, output, currRel)
+  }
+
+  private def getLeftCondition: Expression = {
+    val scanFilters = child match {
+      // Get the filters including the manually pushed down ones.
+      case batchScanTransformer: BatchScanExecTransformer =>
+        batchScanTransformer.filterExprs()
+      case fileScanTransformer: FileSourceScanExecTransformer =>
+        fileScanTransformer.filterExprs()
```
Comment (zzcclp) [inline backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHFilterExecTransformer.scala:125]: modify to
```
case scanTransformer: BasicScanExecTransformer => scanTransformer.filterExprs() 
```

**Review by zzcclp — APPROVED** [review]
LGTM. Velox Ci has passed before, this pr doesn't modify any velox code.

**Pull request merged**